### ISSUE TITLE
metrics diff: reports existing file as not existing

### ISSUE THREAD

**Issue by iksnagreb** [iterative/dvc#5731]
# Bug Report

`metrics diff`: reports existing file as not existing

## Description

Using dvc 2.0.13 to compare metrics of pipeline runs, dvc reports a parameters file (listed in the `vars` section of the `dvc.yaml`) as not existing and quits while the file is clearly existing in the workspace as well as the git HEAD (and other previous commits).

```console
ERROR: failed to show metrics diff - failed to parse 'vars' in 'exp/baseline/dvc.yaml': 'exp/project.yaml' does not exist
```

### Environment information

**Output of `dvc doctor`:**

```console
$ dvc doctor
DVC version: 2.0.13 (pip)
---------------------------------
Platform: Python 3.8.5 on Linux-5.4.0-70-generic-x86_64-with-glibc2.29
Supports: http, https, webdav, webdavs
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/sda1
Caches: local
Remotes: webdavs, webdavs
Workspace directory: ext4 on /dev/sda1
Repo: dvc, git

```

**Additional Information (if any):**

[dvc.yaml ](https://gist.github.com/iksnagreb/c2fc28c1937d6b87340a16440472ce3d)
[verbose error log](https://gist.github.com/iksnagreb/16eefb74463da8af8f944a0bffd5eaec)

https://discord.com/channels/485586884165107732/485596304961962003/825417237333803019
https://discord.com/channels/485586884165107732/485596304961962003/826445437924802651

**Comment by iksnagreb** [comment]
@skshetry dvc seems to be able to see the file:

```
exists locally True
# Global project parameters
project:
  # Project root directory
  root: ../../
  # Project binary/program directory
  bin: ../../bin
  # Data parameters
  data:
    # Root directory
    root: ../../data
  # Experiment parameters
  exp:
    # Root directory
    root: ../

exists in HEAD True
# Global project parameters
project:
  # Project root directory
  root: ../../
  # Project binary/program directory
  bin: ../../bin
  # Data parameters
  data:
    # Root directory
    root: ../../data
  # Experiment parameters
  exp:
    # Root directory
    root: ../
```

**Comment by skshetry** [comment]
Thanks @iksnagreb, found the root cause of this. Seems like the issue is with using `..` in the path, which is not being resolved.

**Comment by skshetry** [comment]
@iksnagreb, could you please try out https://github.com/iterative/dvc/pull/5734? Please install it as follows, thanks.
```console
$ pip install git+https://github.com/skshetry/dvc.git@fix-5731#egg=dvc[webdav]
```

**Comment by skshetry** [comment]
@iksnagreb, this should now be fixed in v2.0.15. Please update.

**Comment by iksnagreb** [comment]
@skshetry Thank you!
